### PR TITLE
Extensible multi validation

### PR DESCRIPTION
**Pull request by ynh-zz**
@fotonobile @bthorben

### CODE REVIEW

File: wsgiservice/resource.py [hunk 1]
```diff
@@ -586,6 +649,18 @@ def get_request_data(self):
         """
         request_data = [self.path_params, self.request.GET]
 
+        # Workaround for BEEKEEPER Android Apps
```
Comment (fotonobile) [inline wsgiservice/resource.py]: This work around should be removed. We fixed it some time ago. May be consult with the mobile team real quick.

File: wsgiservice/resource.py [hunk 2]
```diff
@@ -457,13 +500,12 @@ def convert_param(self, method, param, value):
                 value is invalid for the given method and parameter.
         """
         rules = self._get_validation(method, param)
-        if not rules or not rules.get('convert'):
+        if not rules or not rules.get('convert') or (value is None and not rules['mandatory']):
```
Comment (bthorben) [inline wsgiservice/resource.py]: nicer with at least one named boolean

Comment (bthorben) [inline wsgiservice/resource.py]: I don't know the architecture of this, but it interesting that we do this check _inside_ check_param. I would have expected to not call the method at all for none and not mandatory values. More logic here makes this harder to test.

**Comment by bthorben** [comment]
Any updates on this? Is this mergable?

**Comment by fotonobile** [comment]
We are anyway using this dependency: https://github.com/beekpr/wsgiservice/archive/extensible-multi-validation.zip#egg=wsgiservice-0.4.0.dev2.

So we can definitely merge it and change our dependency.

**Pull request merged**